src/SFA.DAS.AssessorService.Application.Api/Services/OuterApiService.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.AssessorService.Application.Infrastructure.OuterApi;
+using SFA.DAS.AssessorService.Application.Interfaces;
+
+namespace SFA.DAS.AssessorService.Application.Api.Services
+{
+    public class OuterApiService : IOuterApiService
+    {
+        private readonly IOuterApiClient outerApiClient;
+        private readonly ILogger<OuterApiService> logger;
+
+        public OuterApiService(IOuterApiClient outerApiClient, ILogger<OuterApiService> logger)
+        {
+            this.outerApiClient = outerApiClient;
+            this.logger = logger;
+        }
+
+        public async Task<IEnumerable<GetStandardsListItem>> GetActiveStandards()
+        {
+            var response = await outerApiClient.Get<GetStandardsListResponse>(new GetActiveStandardsRequest());
+            return response.Standards;
+        }
+
+        public async Task<IEnumerable<GetStandardByIdResponse>> GetAllStandardDetails(IEnumerable<string> standardUIds)
+        {
+            try
+            {
+                var tasks = standardUIds.Select(id => outerApiClient.Get<GetStandardByIdResponse>(new GetStandardByIdRequest(id))).ToList();
```

Comment:
As mentioned previously I get errors sometimes when running this locally and in Visual Studio. Running from command line seems to mitigate the issue somewhat. Will be interested to see what this is like when in an environment, specifically as to whether connection pooling is required in the course api to deal with the volume of requests.

Reply:
agree